scripts/functions/encode_decode.py

```diff
@@ -0,0 +1,217 @@
+#-------------------------------------------------------------------------------------------------------------------------------
+# By Alexandra Lee
+# (updated May 2019)
+#
+# Encode - decode input data using trained compression model
+#-------------------------------------------------------------------------------------------------------------------------------
+import os
+import sys
+import pandas as pd
+import numpy as np
+from scipy.stats import pearsonr, spearmanr
+from keras.models import model_from_json, load_model
+from keras import metrics, optimizers
+import pickle
+
+from functions import utils
+
+randomState = 123
+from numpy.random import seed
+seed(randomState)
+
+
+def vae_encode_decode_shiftA(sample_data,
+                             model_encoder_file,
+                             model_decoder_file,
+                             weights_encoder_file,
+                             weights_decoder_file,
+                             encoded_dir,
+                             gene_id,
+                             out_dir,
+                             seed_input):
+    """
+    vae_encode_decode_shiftA(sample_data: dataframe,
+                             model_encoder_file: string,
+                             model_decoder_file: string,
+                             weights_encoder_file: string,
+                             weights_decoder_file: string,
+                             encoded_dir: string,
+                             gene_id: string,
+                             out_dir: string,
+                             seed_input: int):
+
+    input:
+
+        sample_data:  Dataframe with gene expression data from subset of samples (around the treshold)
+
+        model_encoder_file: file containing the learned vae encoder model
+
+        model_decoder_file: file containing the learned vae decoder model
+
+        weights_encoder_file: file containing the learned weights associated with the vae encoder model
+
+        weights_decoder_file: file containing the learned weights associated with the vae decoder model
+
+        encoded_dir:  directory to use to output offset vector to 
+
+        gene_id: gene you are using as the "phenotype" to sort samples by 
+
+                 This gene is referred to as "target_gene" in comments below
+
+        out_dir: directory to output predicted gene expression to
+
+        seed_input: random seed
+
+    computation:
+        1.  Sort samples based on the expression level of the target gene defined by the user
+        2.  Sample_data are encoded into VAE latent space
+        3.  Encoded expression is decoded back into gene space
+        4.  This computation is repeated for all samples 
+
+    output: 
+         1. encoded expression profile per sample
+         2. predicted expression profile per sample
+
+    """
+
+    # Output file
+    predict_file = os.path.join(out_dir,
```

Comment:
Just a reminder for the future: 
```python
predict_file = f"{out_dir}/vae_predicted_gene_exp_seed/{str(seed_input)}.txt"
```

Reply:
Thanks.  This one is running py3.5 right now but I might switch it later